### PR TITLE
feat: Add and hookup checkout summary to checkout page

### LINKED ISSUE #208: Wire up actual checkout summary data
@aldeed commented on [Tue Aug 07 2018](https://github.com/reactioncommerce/reaction-component-library/issues/195)

In the starter kit, on the checkout page.

This required building a checkout action for the summary with a simplified version of the mini-cart sans most of its controls.

### PR DESCRIPTION
**Pull request by mikemurray**
Resolves #208
Impact: **minor**
Type: **feature**

## Issue

Add and hookup a cart summary component for the cart page.

## Solution

Using the cart page as a base, condense into a single column layout and with other adjustments to transform it into a cart summary to live in the trailing column of the checkout page.

## Testing
1.  Add items to the cart
2. Go to checkout page
3. Verify that the summary has the proper items and totals

### CODE REVIEW

File: src/components/CheckoutSummary/CheckoutSummary.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      title: "Another Great Product",
+      quantity: 1
+    }
+  ]
+};
+
+test("basic snapshot", () => {
+  const component = renderer.create((
+    <ComponentsProvider value={componentsContext}>
+      <CheckoutSummary
```
Comment (willopez) [inline src/components/CheckoutSummary/CheckoutSummary.test.js]: lets add stubs for these props: 
```
hasMoreCartItems: PropTypes.bool,
    loadMoreCartItems: PropTypes.func,
    onChangeCartItemsQuantity: PropTypes.func,
    onRemoveCartItems: PropTypes.func,
```
Getting an error when I click the remove or increment quantities.

Comment (mikemurray) [reply src/components/CheckoutSummary/CheckoutSummary.test.js]: How are you testing the click in a snapshot test?

Comment (willopez) [reply src/components/CheckoutSummary/CheckoutSummary.test.js]: cart_checkout_error

Just clicking on the elements of the summary

Comment (mikemurray) [reply src/components/CheckoutSummary/CheckoutSummary.test.js]: I see, so it's not the test, its the actual component in the page.

Comment (willopez) [reply src/components/CheckoutSummary/CheckoutSummary.test.js]: Yes, in the actual component in the checkout page component.

**Pull request merged**